### PR TITLE
chore: og image fix

### LINKED ISSUE #305: Support images for social sharing
As the discussion in https://github.com/dart-lang/site-webdev/pull/759 points out, the `snippet_img` isn't used. I suspect the same is true for site-www.

### PR DESCRIPTION
**Pull request by chalin**
- Fixes #305 
- Eliminates all page specific `og:image` settings, be they via `image:` or `snippet_img:`.
- Also adjusts the Gemfile config to eliminate deprecation warnings.

This is the site-www counterpart to https://github.com/dart-lang/site-webdev/pull/799.

**Staged at**: https://dartlang-org-dev.firebaseapp.com

---

The [Open Graph debugger run over the main page](https://developers.facebook.com/tools/debug/sharing/?q=https%3A%2F%2Fdartlang-org-dev.firebaseapp.com%2F) of the staged site shows the following Facebook share preview:
  > screen shot 2017-07-08 at 1 58 40 pm

---

The above replaces the current:
  > screen shot 2017-07-08 at 11 28 00 am

---

cc @Sfshaza @filiph

### CODE REVIEW

**Comment by kwalrath** [comment]
It's still possible to specify page-specific images via the `image:` field in the top YAML, right?

File: src/_includes/head.html [hunk 1]
```diff
@@ -17,14 +19,24 @@
...
 
   <!-- Open Graph -->
   <meta property="og:title" content="{{ page.title }}" />
-  <meta property="og:description" content="{% if page.description | strip_html | strip_newlines | truncate: 160 %}{{ page.description }}{% else if page.excerpt %}{{ page.excerpt | strip_html | strip_newlines | truncate: 160 }}{% else %}{{ site.description | strip_html | strip_newlines | truncate: 160 }}{% endif %}" />
-  <meta property="og:type" content="website" />
-  <meta property="og:url" content="{{ site.url }}{{ page.url }}" />
-  <meta property="og:image" content="{% if page.image %}{{ page.image }}{% else %}{{ site.url }}{% asset_path 'dart-logo-wordmark.png' %}{% endif %}" />
+  <meta property="og:description" content="{{desc}}" />
+  <meta property="og:url" content="{{ page.url | absolute_url }}" />{%
+    assign og_image_asset = page.image_asset | default: layout.image_asset | default: 'dart-logo-for-shares.png' | asset_path %}{%
```
Comment (kwalrath) [inline src/_includes/head.html]: Why is the og:image code for www different from the code for webdev?

**Comment by chalin** [comment]
> It's still possible to specify page-specific images via the image: field in the top YAML, right?

Yes. You can use either `'image:` or `image_asset:`.

File: src/_includes/head.html (hunk 1 above)
Comment (chalin) [inline src/_includes/head.html]: [The code](https://github.com/dart-lang/site-webdev/blob/master/src/_includes/head.html#L28-L30) should be the same now, except that `dart-webdev-logo.png` is used as default value.

Comment (kwalrath) [inline src/_includes/head.html]: Ah, OK. I must've been looking at a slightly older version of the webdev head.html file.

**Comment by kwalrath** [comment]
@filiph is good with this change, so I'm committing it.

**Comment by kwalrath** [comment]
Oh, and thanks @chalin for removing those annoying deprecation warnings!

**Pull request merged**